components/supervisor/pkg/gitpod/gitpod-service.go

```diff
@@ -559,6 +641,10 @@ func (gp *APIoverJSONRPC) GetFeaturedRepositories(ctx context.Context) (res []*W
 
 // GetWorkspace calls getWorkspace on the server
 func (gp *APIoverJSONRPC) GetWorkspace(ctx context.Context, id string) (res *WorkspaceInfo, err error) {
+	if gp == nil {
```

Comment:
@csweichel Could you elaborate on why this might be the case? Especially when we come from [here](https://github.com/gitpod-io/gitpod/blob/c2c0629e37964a3046a929d1e5a6da9c0fa7511b/components/supervisor/pkg/ports/ports-config.go#L120)? I think I'm missing out something important here...

Reply:
The issue is [`createGitpodService`](https://github.com/gitpod-io/gitpod/blob/master/components/supervisor/pkg/supervisor/supervisor.go#L233-L236) which returns nil when there's no token.
At this point its easier to handle the nil case within the functions of `APIoverJSONRPC` rather than outside.

Reply:
I still don't get why [this guard](https://github.com/gitpod-io/gitpod/blob/c2c0629e37964a3046a929d1e5a6da9c0fa7511b/components/supervisor/pkg/ports/ports-config.go#L119) does not catch it, but new guard within the function will do it. 😬 

Is there a way that we can simulate the issue to confirm it? We could always return `nil` from `createGitpodService`  to verify it? Sorry if you already done it.